src/operator/nn/mkldnn/mkldnn_pooling-inl.h

```diff
@@ -104,7 +104,8 @@ class MKLDNNPoolingBwd {
 inline bool SupportMKLDNNPooling(const PoolingParam &param) {
   return param.kernel.ndim() == 2 &&
          (param.pool_type == pool_enum::kMaxPooling ||
-          param.pool_type == pool_enum::kAvgPooling);
+          param.pool_type == pool_enum::kAvgPooling) &&
+         (!param.layout.has_value() || param.layout.value() == mshadow::kNCHW);
```

Comment:
@TaoLv are you tracking this separately?

Reply:
Yes. This change looks good to me for now. Two possible follow-ups:
1. if layout parameter will be removed in the next major release, possibly we will not support NHWC.
2. otherwise, if needed, we will extend MKL-DNN integration to support NHWC. Currently I don't see any necessity for it.

Do you think I can put the suggestion for removing layout into https://github.com/apache/incubator-mxnet/issues/9686?